include/storage/fbw.h

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2017 Nordic Semiconductor ASA
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Public API for flash buffered writes
+ */
+
+#ifndef ZEPHYR_INCLUDE_STORAGE_FBW_H_
+#define ZEPHYR_INCLUDE_STORAGE_FBW_H_
+
+/**
+ * @brief Abstraction over flash buffered writes.
+ *
+ * @defgroup fbw flash buffered write Interface
+ * @{
+ */
+
+#include <stdbool.h>
+#include <storage/flash_map.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct fbw_ctx {
```

Comment:
The API only uses pointers to fbw_ctx, so I'm assuming the structure contents are here because it's caller-allocated.

If so, that should be documented when you're ready to take this out of draft status. Something like this maybe:

```suggestion
/**
 * @brief Context structure for buffered flash writes.
 * Users should treat these structures as opaque values and only interact
 * with them through the below API.
 */
struct fbw_ctx {
```

Reply:
> The API only uses pointers to fbw_ctx, so I'm assuming the structure contents are here because it's caller-allocated.

I'm not sure what you mean by 'structure contents'. But yes, I agree that this should be documented as you propose.

Reply:
> I'm not sure what you mean by 'structure contents'.

I mean showing the members inside of struct fbw_ctx instead of defining it as an opaque type.